src/Microsoft.IdentityModel.Abstractions/EventLogLevel.cs

```diff
@@ -0,0 +1,69 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Defines Event Log Levels.
+    /// </summary>
+    public enum EventLogLevel
+    {
+        /// <summary>
+        /// No level filtering is done on this log level. Log messages of all levels will be logged.
+        /// </summary>
+        LogAlways = 0,
+
```

Comment:
LogAlways doesn't have an exact mapping to AspNetCore's LogLevel enum. We might have to consider avoiding logging messages at EventLevel.LogAlways. Today only the [header information](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Logging/LogHelper.cs#:~:text=headerEntry.EventLevel%20%3D%20EventLevel.LogAlways%3B) is logged at this level.

Reply:
So that means there might be an issue because I'm only finding the header when setting Trace, but really, based on the comment here, the header should always be getting written.

Reply:
Yes there is an issue with the translation from EventLevel to ILogger's LogLevel. ILogger always expects a LogLevel and there isn't an option to always log a given message.
Thank you for capturing it as one of the bugs [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1845).